pkg/controller/core/queue_controller.go

```diff
@@ -47,47 +54,119 @@ func NewQueueReconciler(queues *queue.Manager) *QueueReconciler {
 //+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queues/finalizers,verbs=update
 
 func (r *QueueReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// No-op. All work is done in the event handler.
+	var queueObj kueue.Queue
+	if err := r.client.Get(ctx, req.NamespacedName, &queueObj); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("queue", klog.KObj(&queueObj))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling Queue")
+
+	pendingJobs, err := r.queues.Status(&queueObj)
+	if err != nil {
+		r.log.Error(err, "Failed to retrieve queue status")
+		return ctrl.Result{}, err
+	}
+
+	queueObj.Status.PendingWorkloads = int32(pendingJobs)
+	if err := r.client.Status().Update(ctx, &queueObj); err != nil {
+		r.log.Error(err, "Failed to Update queue")
+		return ctrl.Result{}, err
+	}
 	return ctrl.Result{}, nil
 }
 
 func (r *QueueReconciler) Create(e event.CreateEvent) bool {
-	q := e.Object.(*kueue.Queue)
+	q, match := e.Object.(*kueue.Queue)
+	if !match {
+		// No need to interact with the cache for other objects.
```

Comment:
```suggestion
		// No need to interact with the queue manager for other objects.
```

same in all the instances below